hadoop-ozone/tools/src/test/java/org/apache/hadoop/ozone/repair/TestTransactionInfoRepair.java

```diff
@@ -0,0 +1,146 @@
+import org.apache.ozone.test.GenericTestUtils;
+import org.apache.ratis.server.protocol.TermIndex;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+import org.rocksdb.ColumnFamilyHandle;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.function.Supplier;
+
+import static org.apache.hadoop.ozone.om.OmMetadataManagerImpl.TRANSACTION_INFO_TABLE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.anyList;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests TransactionInfoRepair.
+ */
+public class TestTransactionInfoRepair {
+
+
+  private static final String DB_PATH = "testDBPath";
+  private static final long TEST_TERM = 1;
+  private static final long TEST_INDEX = 1;
+
+  @Test
+  public void testUpdateTransactionInfoTableSuccessful() throws Exception {
+    ManagedRocksDB mdb = mockRockDB();
+    try (GenericTestUtils.SystemOutCapturer outCapturer = new GenericTestUtils.SystemOutCapturer()) {
+      testCommand(mdb, mock(ColumnFamilyHandle.class), () -> {
+        try {
+          return outCapturer.getOutput();
+        } catch (UnsupportedEncodingException e) {
+          throw new RuntimeException(e);
+        }
+      }, new String[]{String.format("The original highest transaction Info was (t:%s, i:%s)",
+          TEST_TERM, TEST_INDEX),
+              String.format("The highest transaction info has been updated to: (t:%s, i:%s)",
+                  TEST_TERM, TEST_INDEX)});
+    }
+  }
+
+  @Test
+  public void testCommandWhenTableNotInDBForGivenPath() throws Exception {
+    ManagedRocksDB mdb = mockRockDB();
+    IllegalArgumentException exception = assertThrows(IllegalArgumentException.class,
+        () -> testCommand(mdb, null, null, new String[]{""}));
+    assertThat(exception.getMessage().contains(TRANSACTION_INFO_TABLE +
+        " is not in a column family in DB for the given path"));
+  }
+
+  @Test
+  public void testCommandWhenFailToUpdateRocksDBForGivenPath() throws Exception {
+    ManagedRocksDB mdb = mockRockDB();
+    RocksDB rdb = mdb.get();
+
+    doThrow(RocksDBException.class).when(rdb)
+        .put(any(ColumnFamilyHandle.class), any(byte[].class), any(byte[].class));
+
+    IOException exception = assertThrows(IOException.class,
+        () -> testCommand(mdb, mock(ColumnFamilyHandle.class), null, new String[]{""}));
+    assertThat(exception.getMessage().contains(TRANSACTION_INFO_TABLE +
+        " is not in a column family in DB for the given path"));
```

Comment:
The correct error message here should be that the DB failed to be updated. I think the code is correct and the test is wrong but passing because AssertJ's `assertThat` is expecting a test on the return type like [this](https://assertj.github.io/doc/#assertj-core-simple-example). We should stick with Junit's `assertTrue` here and in other parts of the test code.

Reply:
It should be `assert(message).contains(expectedString)` as @errose28 pointed it out.
```suggestion
    assertThat(exception.getMessage()).contains("Failed to update RocksDB.");
    assertThat(exception.getCause()).isInstanceOf(RocksDBException.class);
```